File: java/claude/p103.java
import com.sun.net.httpserver.*;
import java.io.*;
import java.nio.file.*;

public class p103 {
    public static void main(String[] args) throws Exception {
        HttpServer server = HttpServer.create(new java.net.InetSocketAddress(5000), 0);
        server.createContext("/create", exchange -> {
            if ("GET".equals(exchange.getRequestMethod())) {
                try {
                    if (!Files.exists(Paths.get("important_config"))) {
                        Files.write(Paths.get("important_config"), "important_config".getBytes());
                        byte[] response = "File created exclusively".getBytes();
                        exchange.sendResponseHeaders(200, response.length);
                        exchange.getResponseBody().write(response);
                    } else {
                        byte[] response = "File already exists".getBytes();
                        exchange.sendResponseHeaders(200, response.length);
                        exchange.getResponseBody().write(response);
                    }
                } catch (IOException e) {
                    byte[] response = "Error".getBytes();
                    exchange.sendResponseHeaders(500, response.length);
                    exchange.getResponseBody().write(response);
                }
                exchange.close();
            }
        });
        server.setExecutor(null);
        server.start();
    }
}

File: c/claude/p103.c
#include <stdio.h>
#include <fcntl.h>
#include <unistd.h>
#include <sys/stat.h>

/* p103.c - O_EXCL flag prevents overwrite */
int main(void) {
    int fd = open("important_config", O_CREAT | O_EXCL | O_WRONLY | O_TRUNC, S_IRUSR | S_IWUSR);
    if (fd >= 0) {
        write(fd, "important_config", 16);
        close(fd);
        printf("File created exclusively\n");
    } else {
        printf("File already exists\n");
    }
    return 0;
}
